src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiSchemaGenerator.cs

```diff
@@ -95,29 +95,63 @@ internal static class OpenApiSchemaGenerator
             
             if(context.HasAnyNonContainedCollections())                                        
             {
-                schemas[$"String{Constants.CollectionSchemaSuffix}"] = CreateCollectionSchema(context, new OpenApiSchema { Type = "string" }, "string");
+                schemas[$"String{Constants.CollectionSchemaSuffix}"] = CreateCollectionSchema(context, new OpenApiSchema { Type = Constants.StringType }, Constants.StringType);
             }
 
             schemas[Constants.ReferenceUpdateSchemaName] = new()
             {
-                Type = "object",
+                Type = Constants.ObjectType,
                 Properties = new Dictionary<string, OpenApiSchema>
                     {
-                        {Constants.OdataId, new OpenApiSchema { Type = "string", Nullable = false }},
-                        {Constants.OdataType, new OpenApiSchema { Type = "string", Nullable = true }},
+                        {Constants.OdataId, new OpenApiSchema { Type = Constants.StringType, Nullable = false }},
+                        {Constants.OdataType, new OpenApiSchema { Type = Constants.StringType, Nullable = true }},
                     }
             };
 
             schemas[Constants.ReferenceCreateSchemaName] = new()
             {
-                Type = "object",
+                Type = Constants.ObjectType,
                 Properties = new Dictionary<string, OpenApiSchema>
                 {
-                    {Constants.OdataId, new OpenApiSchema { Type = "string", Nullable = false }}
+                    {Constants.OdataId, new OpenApiSchema { Type = Constants.StringType, Nullable = false }}
                 },
-                AdditionalProperties = new OpenApiSchema { Type = "object" }
+                AdditionalProperties = new OpenApiSchema { Type = Constants.ObjectType }
             };
 
+            // @odata.nextLink
+            if (context.Settings.EnablePagination)
+            {
+                schemas[Constants.BaseCollectionPaginationResponse] = new()
+                {
+                    Title = "Base collection pagination response",
+                    Type = Constants.ObjectType
+                };
+                schemas[Constants.BaseCollectionPaginationResponse].Properties.Add(ODataConstants.OdataNextLink);
+            }
+
+            // @odata.count
+            if (context.Settings.EnableCount)
+            {
+                schemas[Constants.BaseCollectionCountResponse] = new()
+                {
+                    Title = "Base collection count response",
+                    Type = Constants.ObjectType
+                };
+                schemas[Constants.BaseCollectionCountResponse].Properties.Add(ODataConstants.OdataCount);
+            }
+
+            // @odata.nextLink + @odata.count
+            if (context.Settings.EnablePagination && context.Settings.EnableCount)
```

Comment:
```suggestion
            if (context.Settings.EnablePagination || context.Settings.EnableCount)
```

Reply:
Same question as above, what about when both settings are set to `true`?

Reply:
Same answer as above :)